### PR TITLE
Add ffbs-parker-nextnode

### PR DESCRIPTION
**Pull request by SmithChart**
Thia adds ffbs-parker-nextnode - a package of the *parker*-flavor of Gluon.

This package provides `ebtables`-rules that redirect traffic to the
`localnode` IPs on the node itself.

This is needed in networks where the `localnode` addresses are outside the client network - for example when 
using with `parker`.

In Freifunk Braunschweig, for example, the `localnode` address is `2001:bf7:382:0::1`.
But the IP addresses of routers and clients are in `2001:bf7:381::`.
With this rule traffic to the `localnode` address is always forwarded to the router.

(The service on the router should redirect the client to one of routers public addresses - otherwise the TCP connection
would break when the client roams to another node with the same redirect.)

Previously this package has been managed in
https://gitli.stratum0.org/ffbs/ffbs-packages under the name `gluon-ffbsnext-nextnode`.
Last commit-id: 2241cee3e4be96ded955f6bfd40a42cd9b7e53ac

### CODE REVIEW

File: ffbs-parker-nextnode/files/lib/gluon/ebtables/399-localnode [hunk 1]
```diff
@@ -0,0 +1,12 @@
+local client_bridge = require 'gluon.client_bridge'
+local site = require 'gluon.site'
+local next_node = site.next_node({})
+local macaddr = client_bridge.next_node_macaddr()
+
+if next_node.ip4 then
+        rule('PREROUTING -p IPv4 -d ! ' .. macaddr .. ' --ip-dst ' .. site.next_node.ip4() .. ' -j dnat --to-dst ' .. macaddr .. ' --dnat-target ACCEPT', 'nat')
```
Comment (grische) [inline ffbs-parker-nextnode/files/lib/gluon/ebtables/399-localnode:7]: Is the usage of `next_node.ip4` and `site.next_node.ip4()` intentional? Same for ip6.

File: ffbs-parker-nextnode/Makefile [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+PKG_MAINTAINER:=Chris Fiege <chris@tinyhost.de>
+PKG_LICENSE:=MIT
+
+PKG_BUILD_DIR := $(BUILD_DIR)/$(PKG_NAME)
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/ffbs-parker-nextnode
+  TITLE:=gluon-nextnode config for parker
+  DEPENDS:=+ffbs-parker-nodeconfig
```
Comment (grische) [inline ffbs-parker-nextnode/Makefile:15]: As https://github.com/freifunk-gluon/community-packages/pull/142 is still WIP, is this name fixed?

File: ffbs-parker-nextnode/Makefile [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+endef
+
+define Build/Compile
+endef
+
+define Package/ffbs-parker-nextnode/install
+	$(CP) ./files/* $(1)/
+endef
+
+$(eval $(call BuildPackage,ffbs-parker-nextnode))
```
Comment (grische) [inline ffbs-parker-nextnode/Makefile:38]: Maybe use BuildGluonPackage here

File: ffbs-parker-nextnode/Makefile (hunk 2 above)
Comment (maurerle) [reply ffbs-parker-nextnode/Makefile:15]: I would rather remove the depends here, and add a dependency to ffbs-parker-nextnode on `ffbs-gluon-mesh-vpn`, as this package is quite standalone, but the other can only be used efficiently in combination with this package.

**Comment by blocktrron** [comment]
@rotanid  briefly discussed the compatibility of this package with unpatched gluon - are there plans on upstreaming the necessary core changes to gluon?

**Comment by SmithChart** [comment]
> @rotanid briefly discussed the compatibility of this package with unpatched gluon - are there plans on upstreaming the necessary core changes to gluon?

tl;dr: Yes, we plan to do so.

Long answer: It will take some more steps:

**Make sure `parker` is usable for other communities.** 
To  do so we bring all our packages into community-packages and also start to maintain a [fork of gluon](https://github.com/ffbs/gluon-parker/) that provides a patched base-gluon for others to use.
We at FFBS will keep our gluon modifications away from this in our [Gitlab](https://gitli.stratum0.org/ffbs/ffbs-gluon).
We will also need to move our [Gluon packages fork](https://gitli.stratum0.org/ffbs/packages/-/commits/v2023.2.x-ffbs-next?ref_type=heads ) with modifications to Github, so others can contribute.

This is where we are at now.

**Test the new system, at least in our network** (maybe even in other communities)
This way we can make sure that we did not break anything during refactoring.

**Fix fallout**
Well, I am sure we will break something :grin: 

**Discuss how the patches from our fork can be brought into Gluon**
Most of our changes will probably need some compile-time switches in packages built in gluon. This will very likely take some time. And thb I am not sure if I am the right person to handle this.

File: ffbs-parker-nextnode/Makefile (hunk 2 above)
Comment (SmithChart) [reply ffbs-parker-nextnode/Makefile:15]: I don't see why anyone would use this package without `ffbs-parker-nextnode` (aka the *ffbs-parker-vpn-provider*). But you are right regarding the direction of the dependency.

Comment (grische) [reply ffbs-parker-nextnode/Makefile:15]: Let's change it  then and rename the DEPENDS to the new name `ffbs-mesh-vpn-parker `

Comment (SmithChart) [reply ffbs-parker-nextnode/Makefile]: Droped the `DEPENDS` here and will add it to `ffbs-mesh-vpn-parker`.

File: ffbs-parker-nextnode/files/lib/gluon/ebtables/399-localnode (hunk 1 above)
Comment (SmithChart) [reply ffbs-parker-nextnode/files/lib/gluon/ebtables/399-localnode:7]: It's not really good style. But I do not consider it a problem right now.

File: ffbs-parker-nextnode/Makefile (hunk 3 above)
Comment (SmithChart) [reply ffbs-parker-nextnode/Makefile]: Fixed

**Review by jluebbe — APPROVED** [review]
